src/main/java/nwoolcan/viewmodel/brewery/production/batch/DetailBatchViewModel.java

```diff
@@ -1,7 +1,67 @@
 package nwoolcan.viewmodel.brewery.production.batch;
 
+
+import nwoolcan.model.brewery.batch.Batch;
+import nwoolcan.viewmodel.brewery.production.batch.review.EvaluationViewModel;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
 /**
- * Class representing the detail view model of a batch.
+ * View model representing a detailed step.
  */
-public class DetailBatchViewModel {
+public final class DetailBatchViewModel {
+
+    private final int id;
+    private final BatchInfoViewModel batchInfo;
+    private final List<MasterStepViewModel> steps;
+
+    @Nullable
+    private final EvaluationViewModel review;
+
+    /**
+     * Basic constructor with decorator-like pattern.
+     * @param batch the batch to get the data from.
+     */
+    public DetailBatchViewModel(final Batch batch) {
+        this.id = batch.getId();
+        this.batchInfo = new BatchInfoViewModel(); //TODO change to correct implementation when done
+        this.steps = batch.getSteps().stream().map(MasterStepViewModel::new).collect(Collectors.toList());
+        this.review = batch.getEvaluation().map(EvaluationViewModel::new).orElse(null);
```

Comment:
you probably meant BatchEvaluationViewModel

Reply:
yes sorry